### PR TITLE
Implements localization on Modifiers and everything related to them

### PR DESCRIPTION
**Pull request by Emmayuel80**
This is a **big change** since a lot of things were dependent on the name string, so it needs to be **carefully tested** before merging, I did try to test every modifier before submitting and seemed fine but there's a lot of them so it's possible I missed something.

The most critical bug I found while making this is that sometimes i18n doesn't execute the t method at the right time, so it generates _undefined_ values (Mostly when rendering modifiers created with the ModifierType constructor, some modifiers like AddPokeballModifierType render just fine). 
I added a solution for when that happens, but it was by brute force so the root of the cause it's still a mystery, needs more investigation.

Changelog:
-Added berry en locale
-Added modifier types en/es locale
-Added modifier en/es locale contains miscelaneous descriptions for modifiers
-Added nature en/es locale
-Added type en locale
-Import new localization files into i18n plugin
 - Creates localize function withing various modifier classes
 -Reeplaces enum name strings for berry, nature, battle stats, vouchers, attack type boosters, stat boosters
 -Includes Dire Hit on the TempStatBooster modifier
 -Added a workaround for modifier name and description being undefined at rendering time (not the most optimal, needs more investigation)

### CODE REVIEW

File: src/data/battle-stat.ts [hunk 1]
```diff
@@ -6,25 +8,27 @@ export enum BattleStat {
   SPD,
   ACC,
   EVA,
-  RAND
+  CRIT
```
Comment (Flashfyre) [inline src/data/battle-stat.ts:11]: Why are we changing this enum?

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/data/battle-stat.ts (hunk 1 above)
Comment (Emmayuel80) [reply src/data/battle-stat.ts]: Sorry, forgot to revert that change before pushing, I just did on my latest commit!

File: src/modifier/modifier-type.ts [hunk 2]
```diff
@@ -766,59 +966,59 @@ export const modifierTypes = {
...
 
   TEMP_STAT_BOOSTER: () => new ModifierTypeGenerator((party: Pokemon[], pregenArgs?: any[]) => {
     if (pregenArgs)
       return new TempBattleStatBoosterModifierType(pregenArgs[0] as TempBattleStat);
-    const randTempBattleStat = Utils.randSeedInt(6) as TempBattleStat;
+    const randTempBattleStat = Utils.randSeedInt(7) as TempBattleStat;
     return new TempBattleStatBoosterModifierType(randTempBattleStat);
   }),
-  DIRE_HIT: () => new TempBattleStatBoosterModifierType(TempBattleStat.CRIT),
+/* Now included in TEMP_STAT_BOOSTER  DIRE_HIT: () => return new TempBattleStatBoosterModifierType(TempBattleStat.CRIT), */
```
Comment (Flashfyre) [inline src/modifier/modifier-type.ts:1021]: It's... not supposed to be in temp stat booster? It's supposed to be its own due to rarity tier. Please avoid changing logic when making a localization PR.

**Review by Flashfyre — CHANGES_REQUESTED** [review]

**Review by Flashfyre — CHANGES_REQUESTED** [review]
image
Here's an issue reported by someone doing testing. Make sure this is fixed and that the issue isn't present anywhere else.

File: src/locales/en/berry.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { BerryTranslationEntries } from "#app/plugins/i18n.js";
+
+export const berry: BerryTranslationEntries = {
+    "SITRUS": {
+        "name": "Sitrus",
```
Comment (TeKrop) [inline src/locales/en/berry.ts:5]: As the ` Berry` suffix was removed from `getBerryName` function, shouldn't we specify it in every `name` here ? 🤔

**Comment by TeKrop** [comment]
I would advise to make separate branches/PRs to properly introduce these changes : one for berries, another one for natures, etc.🤔 
IMHO It would make it easier to review and test, and make sure there isn't any hidden bug or regression 😅

**Comment by Emmayuel80** [comment]
I'm currently busy with other stuff from work, will revisit this when I gain some free time again 👀

**Comment by Adri1** [comment]
No news about this?

**Comment by bennybroseph** [comment]
Currently out of date. Most of this has now been added in smaller chunks.

**Pull request closed without merging**